frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -52,86 +49,94 @@ const getIconClass = (status: ClusterUpdateStatus) => {
-    {
-      status === ClusterUpdateStatus.UpdatesAvailable
-        ? <Button bsStyle="link" className="co-m-modal-link" onClick={() => (clusterUpdateModal({cv}))}>
-          <i className={iconClass} aria-hidden={true}></i>
-          &nbsp;
-          {status}
+  if (status === ClusterUpdateStatus.UpdatesAvailable) {
+    return <React.Fragment>
+      <div className="co-update-status">
+        {iconClass && <i className={classNames(iconClass, 'update-pending')} aria-hidden={true}></i>}
+        &nbsp;
+        Update available
+      </div>
+      <div>
+        <Button bsStyle="primary" onClick={() => (clusterUpdateModal({cv}))}>
+          Update now
         </Button>
-        : <span>
-          {iconClass && <i className={iconClass} aria-hidden={true}></i>}
-          &nbsp;
-          {status}
-        </span>
-    }
-  </React.Fragment>;
+      </div>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.Updating) {
+    const updatingCondition = getClusterVersionCondition(cv, ClusterVersionConditionType.Progressing, K8sResourceConditionStatus.True);
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        Working towards
+      </div>
+      <div>updatingCondition.message && `${updatingCondition.message}. `</div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.ErrorRetrieving) {
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        Error retrieving
+      </div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.Failing) {
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        {status}
+      </div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  return <span>
+    {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+    &nbsp;
+    Up to date
+  </span>;
 };
 
-const DesiredVersion: React.SFC<DesiredVersionProps> = ({cv}) => {
+const CurrentVersion: React.SFC<CurrentVersionProps> = ({cv}) => {
   const version = getDesiredClusterVersion(cv);
   return version
     ? <React.Fragment>{version}</React.Fragment>
     : <React.Fragment><i className="pficon pficon-warning-triangle-o" aria-hidden="true" />&nbsp;Unknown</React.Fragment>;
 };
 
+const UpdateLink: React.SFC<CurrentVersionProps> = ({cv}) => {
+  const status = getClusterUpdateStatus(cv);
+  return <React.Fragment>
+    {
+      status === ClusterUpdateStatus.ErrorRetrieving || status === ClusterUpdateStatus.Failing || status === ClusterUpdateStatus.Updating
```

Comment:
This logic looks incorrect. We should show the update link if there are updates available. The other statuses don't matter.

Reply:
@spadgett My understanding was that we want users to be able to choose another version in the event that the update fails. Is this no longer a requirement?

Reply:
It's still a requirement. The *only* thing we need to check is if there are updates available. If there are, we should always give the option to the user.

Reply:
Oh gotcha, I see. Thanks for explaining!

Reply:
OK I understand now. The link moves based on the updating status. I do think this might be confusing for a user :/

Reply:
@bmignano @spadgett How do you want to proceed on this?

Reply:
I think that it's okay that it moves (it also changes from a button to a link) because it's only different for the "Update available" status when the CTA has more importance. However, as an alternative we could say "View **detailed progress** or **update** to another version" so that both links are in the same spot and the update link doesn't move. Thoughts? @spadgett

Reply:
OK we can leave it as-is. Thanks!